src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -321,11 +322,10 @@ BlockIO InterpreterCreateQuery::createDatabase(ASTCreateQuery & create)
 
         if (!load_database_without_tables)
         {
-
             /// We use global context here, because storages lifetime is bigger than query context lifetime
             TablesLoader loader{getContext()->getGlobalContext(), {{database_name, database}}, mode};
-            loader.loadTables();
-            loader.startupTables();
+            waitLoad(currentPoolOr(AsyncLoaderPoolId::Foreground), loader.loadTablesAsync());
+            waitLoad(currentPoolOr(AsyncLoaderPoolId::Foreground), loader.startupTablesAsync());
```

Comment:
Let's support async loading here. It will be useful for tests: we will be able to test async loading using DETACH/ATTACH DATABASE without restarting the server. So it will be possible to write functional tests instead of integration tests

Reply:
Done. What kind of functional test could you suggest with this feature? What interesting things can we do in between attach/detach?